anndata/tests/helpers.py

```diff
@@ -163,6 +163,7 @@ def gen_adata(
         AwkArray,
     ),
     layers_types: "Collection[Type]" = (sparse.csr_matrix, np.ndarray, pd.DataFrame),
+    sparse_fmt: str = "csr",
```

Comment:
Is this different than what `X_type` is doing?

Reply:
yes as the other datatypes that use sparse also need it
```python
obsm = dict(
      array=np.random.random((M, 50)),
      sparse=sparse.random(M, 100, format=sparse_fmt),
      df=gen_typed_df(M, obs_names),
      awk_2d_ragged=gen_awkward((M, None)),
      da=da.random.random((M, 50)),
  )
```